### PR TITLE
Add remark-lint-heading-increment and resolve issues

### LINKED ISSUE #23660: maintenance(docs): Add remark-lint-heading-increment
## Summary

Enable the [`heading-increment`](https://github.com/remarkjs/remark-lint/tree/master/packages/remark-lint-heading-increment) rule to our remark-lint config.

Followup of #21919 

## Motivation

[Headings should only increase 1 level at a time](https://www.w3.org/WAI/tutorials/page-structure/headings/#heading-ranks) for accessibility.

## Steps

1. Delete the [line in `.remarkrc.js`](https://github.com/gatsbyjs/gatsby/blob/master/.remarkrc.js#L16) disabling this configuration.
2. Run `yarn lint:docs`
3. Fix the lint errors

### PR DESCRIPTION
**Pull request by bcye**
## Description
I added the `remark-lint-heading-increment` and fixed all linting errors.

### Documentation
n/a

## Related Issues
Fixes #23660

### CODE REVIEW

File: docs/blog/2020-03-26-service-relief-project/index.md [hunk 1]
```diff
@@ -26,7 +26,7 @@ Wonderfully, though, fundraisers are popping up all over the place to provide so
 - [Building a prototype](#building-a-prototype)
 - [How to Get Involved](#how-to-get-involved)
```
Comment (tesseralis) [inline docs/blog/2020-03-26-service-relief-project/index.md:27]: All these headings should be level 2 (`##`) headings to match the first one.

File: docs/docs/production-app.md [hunk 2]
```diff
@@ -95,15 +95,15 @@ This bundle is produced from [production-app.js](https://github.com/gatsbyjs/gat
 
 This contains the small [webpack-runtime](https://webpack.js.org/concepts/manifest/#runtime) as a separate bundle (configured in `optimization` section). In practice, the app and webpack-runtime are always needed together.
 
-##### framework-[contenthash].js
+#### framework-[contenthash].js
```
Comment (tesseralis) [inline docs/docs/production-app.md:98]: It doesn't make sense to me that these sections are "under" the `webpack-runtime-[contenthash].js` section, but also this page is outdated and I don't think we care too much? What do you think, @sidharthachatterjee

Comment (marcysutton) [reply docs/docs/production-app.md:98]: I agree, this should be an h3 to match the others. Same with the following two headings, right?
 
```suggestion
### framework-[contenthash].js
```

File: docs/docs/production-app.md [hunk 3]
```diff
@@ -95,15 +95,15 @@ This bundle is produced from [production-app.js](https://github.com/gatsbyjs/gat
...
 
 The framework bundle contains the React framework. Based on user behavior, React hardly gets upgraded to a newer version. Creating a separate bundle improves users' browser cache hit rate as this bundle is likely not going to be updated often.
 
-##### commons-[contenthash].js
+#### commons-[contenthash].js
```
Comment (marcysutton) [inline docs/docs/production-app.md:102]:
```suggestion
### commons-[contenthash].js
```

File: docs/docs/production-app.md [hunk 4]
```diff
@@ -95,15 +95,15 @@ This bundle is produced from [production-app.js](https://github.com/gatsbyjs/gat
...
 
 Libraries used on every Gatsby page are bundled into the commons javascript file. By bundling these together, you can make sure your users only need to download this bundle once.
 
-##### component---[name]-[contenthash].js
+#### component---[name]-[contenthash].js
```
Comment (marcysutton) [inline docs/docs/production-app.md:106]:
```suggestion
### component---[name]-[contenthash].js
```

**Comment by marcysutton** [comment]
I think this is good to go now, thanks for the PR!

**Pull request merged**